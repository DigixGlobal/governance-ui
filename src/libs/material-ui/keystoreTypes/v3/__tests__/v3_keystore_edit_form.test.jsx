import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import V3KeystoreEditForm from '../v3_keystore_edit_form';

const formData = {
  name: 'test name',
};

const data = {
  type: 'testType',
  addresses: [{
    name: 'test address 1',
    address: '0x12345',
    tokens: [{
      id: 1,
    }, {
      id: 2,
    }],
    networks: [{
      id: 1,
    }, {
      id: 2,
    }],
  }],
};

describe('<V3KeystoreEditForm />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <V3KeystoreEditForm
        formData={formData}
        formChange={jest.fn}
        setFormData={jest.fn}
        data={data}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
