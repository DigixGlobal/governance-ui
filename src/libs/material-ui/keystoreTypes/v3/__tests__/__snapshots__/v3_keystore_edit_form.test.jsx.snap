// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<V3KeystoreEditForm /> renders correctly 1`] = `
<div>
  <FormField
    formChange={[Function]}
    formData={
      Object {
        "name": "test name",
      }
    }
    label="Name"
    name="name"
    placeholder="Nickname for the Address"
  />
  <Connect(NetworkTokensSelector)
    formChange={[Function]}
    formData={
      Object {
        "name": "test name",
      }
    }
  />
  <V3KeystoreDownloadButton
    keystore={
      Object {
        "addresses": Array [
          Object {
            "address": "0x12345",
            "name": "test address 1",
            "networks": Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
            ],
            "tokens": Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
            ],
          },
        ],
        "type": "testType",
      }
    }
    props={
      Object {
        "color": "blue",
        "content": "Download JSON",
        "icon": "download",
      }
    }
  />
</div>
`;
